src/transformStateWithClones.js

```diff
@@ -66,6 +66,38 @@
  */
 function transformStateWithClones(state, transforms) {
   // write code here
-}
+  const newStates = [];
+  const newState = {
+    ...state,
+  };
+
+  for (const transform of transforms) {
+    switch (transform.operation) {
+      case 'addProperties':
+        for (const key in transform.properties) {
+          newState[key] = transform.properties[key];
+        }
+        // newState = Object.assign(newState, transform.properties);
+        newStates.push(Object.assign({}, newState));
```

Comment:
you can use spread operator as well (see line 70). Better use spread operator and not mixing them.